hibernate-core/src/main/java/org/hibernate/query/sqm/internal/DomainParameterXref.java

```diff
@@ -42,38 +42,9 @@ public static DomainParameterXref from(SqmStatement<?> sqmStatement) {
 		// `.. where a.b = :param or a.c = :param`.  Here we have 2 SqmParameter
 		// references (one for each occurrence of `:param`) both of which map to
 		// the same QueryParameter.
-		final Map<SqmParameter,QueryParameterImplementor<?>> xrefMap = new TreeMap<>(
-				(o1, o2) -> {
-					if ( o1 instanceof SqmNamedParameter ) {
-						final SqmNamedParameter<?> one = (SqmNamedParameter<?>) o1;
-						return o2 instanceof SqmNamedParameter<?>
-								? one.getName().compareTo( ((SqmNamedParameter<?>) o2).getName() )
-								: -1;
-					}
-					else if ( o1 instanceof SqmPositionalParameter ) {
-						final SqmPositionalParameter<?> one = (SqmPositionalParameter<?>) o1;
-						return o2 instanceof SqmPositionalParameter<?>
-								? one.getPosition().compareTo( ( (SqmPositionalParameter<?>) o2 ).getPosition() )
-								: 1;
-					}
-					else if ( o1 instanceof SqmJpaCriteriaParameterWrapper
-							&& o2 instanceof SqmJpaCriteriaParameterWrapper ) {
-//						final SqmJpaCriteriaParameterWrapper wrapper1 = (SqmJpaCriteriaParameterWrapper) o1;
-//						final SqmJpaCriteriaParameterWrapper wrapper2 = (SqmJpaCriteriaParameterWrapper) o2;
-//						if ( wrapper1.getJpaCriteriaParameter() == wrapper2.getJpaCriteriaParameter() ) {
-//							return 0;
-//						}
-//
-//						if ( o1.getName() != null ) {
-//							return o1.getName().compareTo( o2.getName() );
-//						}
-//						else {
-							return Integer.compare( o1.hashCode(), o2.hashCode() );
-//						}
-					}
-
-					throw new HibernateException( "Unexpected SqmParameter type for comparison : " + o1 + " & " + o2 );
-				}
+		final Map<SqmParameter, QueryParameterImplementor<?>> xrefMap = new TreeMap<>(
```

Comment:
I think we should question the use of `TreeMap` here in general. Why is this not a simple `HashMap`?

Reply:
Because the TreeMap does deduplication.  If we can still dedup with a HashMap, then sure

Reply:
Now, I'm not sure why that commented code got commented out and I think that may be part of the problem here.

However, the approach here is perfectly fine also.

Reply:
Though we should probably make `SqmParameter` as `Comparable` and drop the lambda in the `TreeMap` ctor